### ISSUE TITLE
[LINT_BUG]: box_alphabetical_calls_linter behaves differently on different OSes

### ISSUE THREAD

**Issue by DeepanshKhurana** [Appsilon/box.linters#168]
### box.linters version

`0.10.5`

```
> packageVersion("box.linters")
[1] �\u0080\u00980.10.5�\u0080\u0099
```

### Sample source code to lint

```r
box::use(
  DT[
    DTOutput,
    datatable,
    renderDT
  ],
  bslib[
    card,
    card_header,
    layout_column_wrap,
  ],
  config[
    get
  ],
  dplyr[
    arrange,
    desc,
    filter,
    mutate,
    select,
  ],
  echarts4r[
    echarts4rOutput,
    renderEcharts4r
  ],
  graphics[
    plot.new,
    text
  ],
  lubridate[
    days,
    minutes
  ],
  shiny,
  shinycssloaders[
    withSpinner
  ],
)
```

```r
box::use(
  FactoMineR[
    PCA
  ],
  bslib[
    card,
    card_header,
    layout_column_wrap
  ],
  dplyr[
    select
  ],
  factoextra[
    fviz_contrib
  ],
  ggplot2[
    aes,
    geom_bar,
    geom_line,
    geom_point,
    ggplot,
    ggtitle,
    labs,
    theme_minimal,
  ],
  shiny,
  shinycssloaders[
    withSpinner
  ],
)
```

### Lint command used

lintr::lint(
  code,
  linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters)
)

### Lint result

`box_alphabetical_calls_linter` triggers for the uppercase function and module names.

### Expected result

There should be no lint errors.

**Comment by DeepanshKhurana** [comment]
For further information and context, we've already discussed this on the [tech-team channel On Slack](https://appsilon.slack.com/archives/C0J5ATBLH/p1742800404025479). 

```r
> sort(c("A", "B", "C", "a", "b", "c"))
[1] "A" "B" "C" "a" "b" "c"
> sort(c("A", "B", "C", "a", "b", "c"), method = "shell")
[1] "A" "B" "C" "a" "b" "c"
> sort(c("A", "B", "C", "a", "b", "c"), method = "radix")
[1] "A" "B" "C" "a" "b" "c"
> sort(c("A", "B", "C", "a", "b", "c"), method = "quick")
[1] "A" "B" "C" "a" "b" "c"
```

The conclusion @radbasa drew is

> Must be why {renv} uses ASCII sort

The issues is simply that Windows does not sort things in the natural order i.e. does not equate `e` and `E` to be the same. It uses ASCII values and is case-sensitive.

**Comment by DeepanshKhurana** [comment]
I think `method = "radix"` is good and consistent across operating systems. It can work _iff_ a user is made aware of how things work i.e. 

> A-Z comes before a-z

From what I can test with the code block you shared from `renv`, at least on Windows and Mac, the results from this sort are identical to `radix` sort. So, we can take the simpler approach and use `method = "radix"` and call it a day.

It also works really well with numbers and periods in the mix not that we should mostly face that use-case.

```r
> sort(c("A", "B", "C.r", "Ab1", "a", "b", "c.A"), method = "radix")
[1] "A"   "Ab1" "B"   "C.r" "a"   "b"   "c.A"
```

**Comment by radbasa** [comment]
```
# MacOS
> sort(c("A", "B", "C.r", "Ab1", "a", "b", "c.A"), method = "radix")
[1] "A"   "Ab1" "B"   "C.r" "a"   "b"   "c.A"

# Linux
> sort(c("A", "B", "C.r", "Ab1", "a", "b", "c.A"), method = "radix")
[1] "A"   "Ab1" "B"   "C.r" "a"   "b"   "c.A"
```

Seems to be consistent enough
